client/store/singleProductSlice.js

```diff
@@ -22,6 +31,9 @@ const singleProductSlice = createSlice({
         builder.addCase(fetchSingleProduct.fulfilled, (state, action) => {
             state.singleProduct = action.payload
         })
+        builder.addCase(addNewProduct.fulfilled, (state, action) => {
+            state.push(action.payload)
```

Comment:
Have you tested this? I dont know if this will work given that state is an object & push is an array method

Also - this redux slice is used only for the single product page, not sure this is the place to incorporate adding new products?